replica/database.cc

```diff
@@ -1025,12 +1025,19 @@ future<> database::add_column_family(keyspace& ks, schema_ptr schema, column_fam
     if (_ks_cf_to_uuid.contains(kscf)) {
         throw std::invalid_argument("Column family " + schema->cf_name() + " exists");
     }
-    ks.add_or_update_column_family(schema);
-    schema->registry_entry()->set_table(cf->weak_from_this());
     auto holder = co_await _cf_lock.hold_write_lock();
-    _column_families.emplace(uuid, std::move(cf));
-    _ks_cf_to_uuid.emplace(std::move(kscf), uuid);
-    _column_families[uuid]->start();
+    try {
+        ks.add_or_update_column_family(schema);
+        schema->registry_entry()->set_table(cf->weak_from_this());
+        _column_families.emplace(uuid, std::move(cf));
+        _ks_cf_to_uuid.emplace(std::move(kscf), uuid);
+        _column_families[uuid]->start();
+    } catch (...) {
+        _ks_cf_to_uuid.erase(std::make_pair(schema->ks_name(), schema->cf_name()));
+        _column_families.erase(uuid);
+        ks.metadata()->remove_column_family(schema);
+        throw;
+    }
```

Comment:
This should fix a bug (open one if there isn't any already) so we can consider backporting

Reply:
what about rolling back the schema registry?

Reply:
It sets a weak ptr to cf, so it will just unregister itself